### PR TITLE
Multitenant payment processor configuration

### PR DESCRIPTION
**Pull request by e-kolpakov**
**Description:** This PR reintroduces changes made in #630, but without startup validation that caused [some trouble](https://openedx.atlassian.net/browse/WL-390)
**Merge deadline:** 8 Apr
**Related:** [Discovery document](https://docs.google.com/document/d/17BRYvkO4ETvBV9TFodElpbJaK4oze7e0mNqf9ztnGnA/edit#heading=h.70eeasm5mr1q)
**Sandbox:** 
- edX site: [LMS](http://pr11898.sandbox.opencraft.com), [Studio](http://studio.pr11898.sandbox.opencraft.com), [eCommerce](https://ecommerce.pr11898.sandbox.opencraft.com)
- WL site: [LMS](http://wlsite.pr11898.sandbox.opencraft.com), [eCommerce](https://ecommerce.wlsite.pr11898.sandbox.opencraft.com)

**JIRA:** [WL-362](https://openedx.atlassian.net/browse/WL-362), [WL-390](https://openedx.atlassian.net/browse/WL-390)

**Testing instructions:** See #630

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @e-kolpakov! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=654&repo=edx%2Fecommerce

**Comment by e-kolpakov** [comment]
@clintonb @douglashall Please review. This PR contains main body of changes added in #630, and lacks startup validation feature that caused trouble - it is moved to a separate PR (#655)

File: ecommerce/core/config.py [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 from django.apps import AppConfig
 
+log = logging.getLogger(__name__)
+
 
 class CoreAppConfig(AppConfig):
     name = 'ecommerce.core'
     verbose_name = 'Core'
 
+    OPERATIONAL_ERROR_MESSAGE = 'DB error when validating configuration - most likely DB was not created yet - skipping'
```
Comment (jbzdak) [inline ecommerce/core/config.py]: This is used in the other PR, and in this PR it is a dead code. Please consider moving it.

File: ecommerce/core/models.py [hunk 2]
```diff
@@ -45,6 +55,68 @@ class SiteConfiguration(models.Model):
...
+    def _clean_payment_processors(self):
+        """
+        Validates payment_processors field value
+
+        Raises:
+            ValidationError: If `payment_processors` field contains invalid/unknown payment_processor names
+        """
+        value = self.payment_processors.strip()
+        if not value:
+            raise ValidationError('Invalid payment processors field: must not only contain whitespace characters')
```
Comment (jbzdak) [inline ecommerce/core/models.py]: NIT: Please consider changing: "must not only contain whitespace" to "must contain some text" or "cannot be empty".

**Comment by jbzdak** [comment]
Two nits, otherwise +1.

**Comment by douglashall** [comment]
:+1:

**Comment by e-kolpakov** [comment]
@clintonb friendly reminder - please review.

**Comment by clintonb** [comment]
I'm not reviewing this a second (third?) time. What changed?

**Comment by e-kolpakov** [comment]
@clintonb tl;dr nothing changed, but I still need your formal approval to adhere to rules edX set up on merging PRs.

This is a new PR and, according to edX PR rules, I've got to get explicit approvals in form of :+1: emoticon from at least one edX team member, or if the change touches areas of responsibility of two or more teams - one per team.

I believe this "White Label Analytics" project is handled by solutions team but this particular PR touches ecommerce, thus, to adhere to formal rules, I've got to get approvals from solutions and ecommerce team members. @douglashall already provided "solutions" approval, so now I need "ecommerce" approval. You've been the reviewer for original PR, so it makes sense for you to review it. If you don't have time to do that - please assign someone else from ecommerce team to do that.

**Comment by clintonb** [comment]
:+1: I don't have time to review this again. I barely had time the first time. In the future we must discuss all of the alternatives before reverting PRs wholesale. The fix for the startup issue was literally a one line change that I recommended before the revert.

**Comment by e-kolpakov** [comment]
@clintonb thank you. There's second PR for re-adding startup validation #655 - who do I talk to to get it approved from ecommerce team? the code is the same there as well, just broader exception is caught.

**Comment by douglashall** [comment]
@clintonb Agreed on agreeing about wholesale reverts. You and I had agreed to revert a smaller change, but the full revert was merged instead. https://github.com/edx/ecommerce/pull/651

I would encourage you to socialize this with the rest of the ecommerce team so that we can improve our communication in the future.

**Pull request merged**